pkg/resources/installPlan.go

```diff
@@ -0,0 +1,34 @@
+package resources
+
+import (
+	"context"
+	"github.com/operator-framework/operator-lifecycle-manager/pkg/api/apis/operators/v1alpha1"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	pkgclient "sigs.k8s.io/controller-runtime/pkg/client"
+	"strings"
+)
+
+func upgradeApproval(ctx context.Context, client pkgclient.Client, ip *v1alpha1.InstallPlan, maxVersion *Version) error {
+	if ip.Spec.Approved == false && len(ip.Spec.ClusterServiceVersionNames) > 0 {
+		logrus.Infof("getting version for %s", ip.Spec.ClusterServiceVersionNames[0])
+		// convert "keycloak.1.8.2" into "1.8.2"
+		verStr := strings.SplitN(ip.Spec.ClusterServiceVersionNames[0], ".", 2)[1]
```

Comment:
this seems a bit gross. How do we know it will always be at index 0?

Reply:
because we make them, it is really gross. I could not see a cleaner way to get the version number though, as the version in this object is the likes of "v1alpha1".